rpc/did/runtime-api/src/lib.rs

```diff
@@ -0,0 +1,107 @@
+// KILT Blockchain – https://botlabs.org
+// Copyright (C) 2019-2022 BOTLabs GmbH
+
+// The KILT Blockchain is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The KILT Blockchain is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+// If you feel like getting in touch with us, you can do so at info@botlabs.org
+#![cfg_attr(not(feature = "std"), no_std)]
+
+#[cfg(feature = "std")]
+use serde::{Deserialize, Serialize};
+
+use codec::{Codec, Decode, Encode, MaxEncodedLen};
+use scale_info::TypeInfo;
+use sp_std::vec::Vec;
+
+mod did_details;
+mod service_endpoint;
+
+pub use did_details::*;
+pub use service_endpoint::*;
+
+#[derive(Encode, Decode, TypeInfo, PartialEq)]
+#[cfg_attr(feature = "std", derive(serde::Serialize, serde::Deserialize))]
+#[cfg_attr(
+	feature = "std",
+	serde(bound(
+		serialize = "
+		Balance: std::fmt::Display,
+		AccountId: Serialize,
+		Key: Serialize,
+		BlockNumber: Serialize,
+		DidIdentifier: Serialize,
+		Type: Serialize,
+		Url: Serialize,
+		Id: Serialize,
+		Web3Name: Serialize,",
+		deserialize = "
+		Balance: std::str::FromStr,
+		AccountId: Deserialize<'de>,
+		Key: Deserialize<'de>,
+		BlockNumber: Deserialize<'de>,
+		DidIdentifier: Deserialize<'de>,
+		Type: Deserialize<'de>,
+		Url: Deserialize<'de>,
+		Id: Deserialize<'de>,
+		Web3Name: Deserialize<'de>,"
+	))
+)]
+pub struct DidDocument<DidIdentifier, AccountId, Web3Name, Id, Type, Url, Balance, Key: Ord, BlockNumber: MaxEncodedLen>
```

Comment:
I am not 100% sure `DidDocument` is the best name here, considering it already has a meaning on its own. `DidDetails` of course does not sound nice either. What about something like `DidLinkedInfo`?

Reply:
yeah wasn't sure myself. `DidLinkedInfo` sounds like it's the account linking. But I think it's already better than the DidDoc. What do you think about `DidRecord`, `DidListing`, `DidEntry`, `DidData`

Reply:
ccc688f9c366ffe277ca6c47849885b94aff14d6

I renamed it to `DidLinkedInfo`